test/plugin.test.ts

```diff
@@ -0,0 +1,178 @@
+import * as jsonschema from "jsonschema";
+import * as jsonfile from "jsonfile";
+import fetch from "node-fetch";
+
+import { structure, imageFormats } from "./standard";
+import {
+  checkFileChecksum,
+  getZipFileList,
+  getImageDimensions,
+  isVersionBiggerThan,
+} from "./utils";
+
+const storeListURL = "https://raw.githubusercontent.com/massalabs/thyra-plugin-store/main/plugins.json"
+
+describe("TestFormat", () => {
+  let plugins: any[];
+  let lastVersion: any[];
+
+  beforeAll(async () => {
+    plugins = await jsonfile.readFile("plugins.json");
+    lastVersion = await fetch(storeListURL).then((res) => res.json());
+  });
+  test("plugins is list", () => {
+    expect(plugins).toBeInstanceOf(Array);
+  });
+
+  test("elements follows mandatory structure", () => {
+    for (const data of plugins) {
+      const result = jsonschema.validate(data, structure);
+      expect(result.errors).toHaveLength(0);
+    }
+  });
+  test("if one plugin has been added, all the other are exactly the same", () => {
+    if (plugins.length > lastVersion.length) {
+      for (const data of lastVersion) {
+        expect(plugins).toContainEqual(data);
+      }
+    }
+  });
+  test("one plugin is updated if there is an update", () => {
+    if (plugins.length == lastVersion.length) {
+      // get the number of plugin where the version has been updated in one line
+      const updatedPlugins = plugins.filter(
+        (plugin, index) =>
+          plugin.version != lastVersion[index].version &&
+          plugin.name == lastVersion[index].name
+      );
+      const updatedPluginsAmount = updatedPlugins.length;
+      expect(updatedPluginsAmount).toBe(1);
+      if (updatedPluginsAmount == 1) {
+        // get the plugin that has been updated in the last version
+        const lastVersionUpdatedPlugin = lastVersion.filter(
+          (plugin, index) => plugin.version != plugins[index].version
+        )[0];
```

Comment:
Could you, please, double check of all your filters!? I am seeing on each test. I have no context but I have the feeling they are repeated. 

About array position `[0]` if we could be more declarative would be better. It can be tricky. I guess in that case the`slice()` method does the job. Isn't it!? 🤔